src/vs/platform/terminal/common/terminal.ts

```diff
@@ -797,6 +797,16 @@ export interface ITerminalProfileSource extends IBaseUnresolvedTerminalProfile {
 
 export interface ITerminalContributions {
 	profiles?: ITerminalProfileContribution[];
+	quickFixes?: ITerminalQuickFixContribution[];
+}
+
+export interface ITerminalQuickFixContribution {
+	id: string;
+	commandLineMatcher: string | RegExp;
+	outputMatcher: ITerminalOutputMatcher;
+	exitStatus?: boolean;
+	commandToRun?: string;
```

Comment:
todo: add to docs that only `commandToRun` or `linkToOpen` will be supported for a given quick fix - not both. `commandToRun` when provided will be used